frontend/pluginloader.lua

```diff
@@ -55,23 +55,45 @@ local function getMenuTable(plugin)
     return t
 end
 
+-- Event handlers defined by plugins are wrapped in a HandlerSandbox
+-- The purpose of the sandbox is to get meaningful stack-traces out of errors happening in plugins.
+local HandlerSandbox = { mt = {} }
+
+function HandlerSandbox.new(context, fname, f, module)
+    local t = {
+        context = context,
+        fname = fname,
+        f = f,
+    }
+    return setmetatable(t, HandlerSandbox.mt)
+end
+
+function HandlerSandbox:call(...)
+    local traceback = function (err)
+         -- do not print 2 topmost entries in traceback. The first is this local function
+         -- and the second is the call method of HandlerSandbox.
+         logger.err("An error occurred while executing a handler:\n" ..err .. "\n" .. debug.traceback(self.context.name .. ":" .. self.fname, 2))
+    end
+    local ok, re = xpcall(self.f, traceback, self.context, unpack(arg)) -- unpack arg table into arguments
```

Comment:
Anything wrong with keeping a clean variadic `...` forward?

As is, this breaks if there are `nil`s in the arguments (`unpack` *can* help with that, but only when using the three-argument form, where you pass the start & ends indices, c.f., `frontend/ui/widget/eventlistener.lua`).

Reply:
Was not aware of the variadic forwarding. Is this a LuaJIT extension? I'll update to use that.